service/dap/server.go

```diff
@@ -882,22 +882,27 @@ func (s *Server) onStepOutRequest(request *dap.StepOutRequest) {
 
 func (s *Server) doStepCommand(command string, threadId int) {
 	// Use SwitchGoroutine to change the current goroutine.
-	state, err := s.debugger.Command(&api.DebuggerCommand{Name: api.SwitchGoroutine, GoroutineID: threadId}, nil)
-	if err != nil {
-		s.log.Errorf("Error switching goroutines while stepping: %e", err)
-		// If we encounter an error, we will have to send a stopped event
-		// since we already sent the step response.
-		stopped := &dap.StoppedEvent{Event: *newEvent("stopped")}
-		stopped.Body.AllThreadsStopped = true
-		if state.SelectedGoroutine != nil {
-			stopped.Body.ThreadId = state.SelectedGoroutine.ID
-		} else if state.CurrentThread != nil {
-			stopped.Body.ThreadId = state.CurrentThread.GoroutineID
+	var err error
+	if threadId != 0 {
+		_, err = s.debugger.Command(&api.DebuggerCommand{Name: api.SwitchGoroutine, GoroutineID: threadId}, nil)
+		if err != nil {
+			s.log.Errorf("Error switching goroutines while stepping: %e", err)
+			// If we encounter an error, we will have to send a stopped event
+			// since we already sent the step response.
+			stopped := &dap.StoppedEvent{Event: *newEvent("stopped")}
+			stopped.Body.AllThreadsStopped = true
+			if state, err := s.debugger.State(false); err != nil {
```

Comment:
why State(false)?

Reply:
The Command call immediately preceeding this will block until the target stops, there is no reason to do a non-blocking state call here.